common-api/src/main/java/org/pmiops/workbench/cdr/dao/DbDomainDao.java

```diff
@@ -0,0 +1,70 @@
+
+        @Query(nativeQuery=true,value="select d.domain_id, d.domain_display, d.domain_desc,\n" +
+                "d.db_type, d.domain_route,d.concept_id, count(distinct c.concept_id) as standard_concept_count\n" +
+                "from db_domain d\n" +
+                "join concept c on d.domain_id = c.domain_id\n" +
+                "left join concept_synonym cs on c.concept_id=cs.concept_id \n" +
+                "where d.db_type = 'domain_filter' and (c.count_value > 0 or c.source_count_value > 0) \n" +
+                "and  ((((match(c.concept_name) against(?1 in boolean mode) ) or\n" +
+                "(match(cs.concept_synonym_name) against(?1 in boolean mode))) and\n" +
+                "c.standard_concept IN ('S', 'C') or (c.concept_id=?2 or c.concept_code=?2))\n" +
+                "group by d.domain_id, d.domain_display, d.domain_desc, d.db_type, d.concept_id\n" +
+                " \n" +
+                "union\n" +
+                " \n" +
+                "select d.domain_id, d.domain_display, d.domain_desc, d.db_type, d.domain_route,\n" +
+                "d.concept_id, count(distinct q.concept_id) as standard_concept_count from db_domain d\n" +
+                "join achilles_results r on d.concept_id = r.stratum_1\n" +
+                "join concept q on r.stratum_2 = q.concept_id\n" +
+                "left join concept_synonym cs on q.concept_id=cs.concept_id \n" +
+                "where d.db_type = 'survey' and r.analysis_id = 3110\n" +
+                "and (((match(q.concept_name) against(?1 in boolean mode) ) or\n" +
+                "(match(cs.concept_synonym_name) against(?1 in boolean mode) ) or\n" +
+                "(q.concept_id=?2 or q.concept_code=?2)) or\n" +
+                "(match(r.stratum_4) against(?1 in boolean mode)))\n" +
+                "group by d.domain_id, d.domain_display, d.domain_desc, d.db_type, d.concept_id \n ")
+    List<DbDomain> findDomainSearchResults(String keyword,String query);
+
+    @Query(nativeQuery=true,value="select d.domain_id, d.domain_display, d.domain_desc, d.db_type,\n" +
+            "d.domain_route, d.concept_id, count(c.concept_id) as standard_concept_count from db_domain d\n" +
+            "join concept c on d.domain_id = c.domain_id\n" +
+            "where d.db_type = 'domain_filter' and c.count_value > 0 and c.standard_concept IN ('S', 'C')\n" +
+            "group by d.domain_id, d.domain_display, d.domain_desc, d.db_type, d.concept_id\n" +
+            " \n" +
+            "union\n" +
+            " \n" +
+            "select d.domain_id, d.domain_display, d.domain_desc, d.db_type, d.domain_route, d.concept_id,\n" +
+            "count(q.concept_id) as standard_concept_count from db_domain d\n" +
+            "join achilles_results r on d.concept_id = r.stratum_1\n" +
+            "join concept q on r.stratum_2 = q.concept_id\n" +
+            "where d.db_type = 'survey' and r.analysis_id = 3110\n" +
```

Comment:
This is probably an existing issue, but what's up with these extra criteria thrown in here? Is this answering a much more specific question than advertised by the DAO method name??